source/util/CommandLine.cpp

```diff
@@ -492,11 +536,24 @@ std::string CommandLine::getHelpText(string_view overview) const {
     }
 
     // Finally append all groups to the output.
+    std::string indent = fmt::format("  {:{}}"sv, " "sv, maxLen);
     for (auto& [opt, key] : lines) {
         result += fmt::format("  {:{}}"sv, key, maxLen);
-        if (!opt->desc.empty())
-            result += fmt::format("  {}", opt->desc);
-        result += "\n";
+        if (!opt->desc.empty()) {
+            bool first = true;
+            std::istringstream iss(opt->desc);
+            std::string item;
+            while (std::getline(iss, item, '\n')) {
+                if (!first)
```

Comment:
I think this drops the spacing between options and their descriptions? I think if you run the unit tests they should catch this. It would be good to update them for the new features too.

Reply:
It doesn't (see quote in the original PR description, it's copied directly from the help screen).
Here is a larger snippet from the `--help` output:
```
  --libraries-inherit-macros               If true, library files will inherit macro definitions from the primary source files. --single-unit must also be passed when this option is used.
  --cmd_ignore <vendor_cmd>,<N>            Define rule to ignore vendor command <vendor_cmd> with its following <N> parameters.
                                           A command of the form +xyz will also match any vendor command of the form +xyz+abc,
                                           as +abc is the command's argument, and doesn't need to be matched.
  --cmd_rename <vendor_cmd>,<slang_cmd>    Define rule to rename vendor command <vendor_cmd> into existing <slang_cmd>
  --max-parse-depth <depth>                Maximum depth of nested language constructs allowed
  --max-lexer-errors <count>               Maximum number of errors that can occur during lexing before the rest of the file is skipped
  --max-hierarchy-depth <depth>            Maximum depth of the design hierarchy
  --max-generate-steps <steps>             Maximum number of steps that can occur during generate block evaluation before giving up
  --max-constexpr-depth <depth>            Maximum depth of a constant evaluation call stack
  --max-constexpr-steps <steps>            Maximum number of steps that can occur during constant evaluation before giving up
  --constexpr-backtrace-limit <limit>      Maximum number of frames to show when printing a constant evaluation backtrace; the rest will be abbreviated
  --max-instance-array <limit>             Maximum number of instances allowed in a single instance array
 ```

Unless you meant something else, which I missed.

This should have been split to a standalone PR, but by the time I realized I needed it, I was already knee deep in the main PR, so I left it as part of the main PR.

Reply:
Oh hmm, previously the code would do += format("  {}", desc), so there was two spaces added. Now the desc doesn't get those two spaces added before it (unless I'm missing something) so I would expect that the very longest option name would no longer have the space. All of the shorter names of course still benefit from the maxLen indentation.

Reply:
Yup, you're right. I see what you mean and upon close examination, the longest one is indeed touching the description. I'll just change the alignment width factor to `maxLen + 1` or `maxLen + 2`.